ui/raid/import_export.ts

```diff
@@ -142,301 +136,799 @@ class RaidWCLImporter extends Importer {
+		this.simUI.setBuffBots(TypedEvent.nextEventID(), buffBots);
+		
+		if (!experimentalGenerateParties) {
+			const rateLimitEnd: wclRateLimitData = await this.getRateLimit(token);
+			console.debug(`Rate Limit resets in ${rateLimitEnd.pointsResetIn} seconds.`);
+		}
+		this.close();
+	}
+}
 
-					// Innervate!
-					if (buff.abilityGameID == 29166) {
-						// Find if target is a player.
-						// If so, find source, could be a player or a buffbot.
-						// Apply innervate.
-						var sourceID = wclIDtoRaidIndex.get(buff.sourceID)!;
-						var targetID = wclIDtoRaidIndex.get(buff.targetID)!;
-						var source = settings.raid!.parties[Math.floor(sourceID/5)].players[sourceID%5];
-						var target = settings.raid!.parties[Math.floor(targetID/5)].players[targetID%5];
-						if (target.class == undefined) {
-							// Target is not a player
-							return;
-						}
-						if (source.class != Class.ClassDruid) {
-							// its a buffbot
-							buffBots.forEach(bot => {
-								if (bot.raidIndex == sourceID) {
-									// Found our buffer
-									bot.innervateAssignment = RaidTarget.create();
-									bot.innervateAssignment.targetIndex = targetID;
-									return;
-								}
-							});
-						} else {
-							// Assign player sources
-							if (source.spec.oneofKind == "balanceDruid") {
-								source.spec.balanceDruid.options!.innervateTarget = RaidTarget.create();
-								source.spec.balanceDruid.options!.innervateTarget.targetIndex = targetID;
-							} else if (source.spec.oneofKind == "feralDruid") {
-								source.spec.feralDruid.options!.innervateTarget = RaidTarget.create();
-								source.spec.feralDruid.options!.innervateTarget.targetIndex = targetID;
-							} else if (source.spec.oneofKind == "feralTankDruid") {
-								source.spec.feralTankDruid.options!.innervateTarget = RaidTarget.create();
-								source.spec.feralTankDruid.options!.innervateTarget.targetIndex = targetID;
-							}
-						}
+class WCLSimPlayer implements wclSimPlayer {
+	public gear: wclGear[];
+	public icon: string;
+	public id: number;
+	public name: string;
+	public type: string;
+	public talents: wclTalents[];
+	public wclSpec: string;
+	
+	public partyAssigned: boolean = false;
+	public partyFound: boolean = false;
+	public partyMembers: string[] = [];
+	
+	public isPartyBuffer: boolean = false;
+	public isBuffBot: boolean = false;
+	public sortPriority: number = 99;
+	
+	private simUI: RaidSimUI;
+	private spec: Spec;
+	private specType: string;
+	private faction: Faction;
+	
+	constructor(data: wclPlayer, simUI: RaidSimUI, faction: Faction = Faction.Unknown) {
+		this.simUI = simUI;
+		
+		this.name = data.name;
+		this.gear = data.gear;
+		this.icon = data.icon;
+		this.id = data.id;
+		this.type = data.type;
+		this.talents = data.talents;
+		this.wclSpec = data.icon.split("-")[1];
+		this.faction = faction;
+		
+		// Prot Paladin's occasionally have a specType of "Protection" instead of "Justicar"?
```

Comment:
Justicar means Prot > Ret but you have at least 21 points in Ret. (41+ in prot means 'Protection')

Reply:
like bear warden vs gaurdian

Reply:
Just let me know when you get it merged in, and I can alter the PR to accommodate it.